doc/release/3.0.0.rst

```diff
@@ -0,0 +1,652 @@
+    ---
+    - [1000, '9990000000000', 'New awesome book', 2023]
+    ...
+
+You can also provide a custom logic for generating a default value.
+To achieve this, create a function using ``box.schema.func.create``:
+
+.. code-block:: lua
+
+    box.schema.func.create('current_year', {
+        language = 'Lua',
+        body = "function() return require('datetime').now().year end"
+    })
+
+Then, assign the function name to ``default_func`` when defining a space format:
+
+.. code-block:: lua
+
+    box.space.books:format({
+        -- ... --
+        { name = 'year_of_publication', type = 'unsigned', default_func = 'current_year' }
+    })
+
+
+
+.. _3-0-triggers:
+
+Triggers
+~~~~~~~~
+
+In the 3.0 version, the API for creating :ref:`triggers <triggers>` is completely reworked.
+A new ``trigger`` module is introduced, allowing you to set handlers on both predefined and custom events.
+
+To create the trigger, you need to:
+
+1)  Provide an event name used to associate the trigger with.
+2)  Define the trigger name.
+3)  Provide a trigger handler function.
+
+The code snippet below shows how to subscribe to changes in the ``books`` space:
+
+.. code-block:: lua
+
+    local trigger = require('trigger')
+    trigger.set(
+            'box.space.books.on_replace', -- event name
+            'some-custom-trigger',        -- trigger name
+            function(...)
+                -- trigger handler
+            end
+    )
+
+
+
+.. _3-0-pagination_for_read_views:
+
+Pagination in read views (EE)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The 2.11 release introduced the following features:
+
+-   :ref:`Read views <2-11-read_views>` are in-memory snapshots of the entire database that aren't affected by future data modifications.
+-   :ref:`Pagination <2-11-pagination>` for getting data in chunks.
+
+With this release, a read view object supports the ``after`` and ``fetch_pos`` arguments for the ``select`` and ``pairs`` methods:
+
+.. code-block:: tarantoolsession
+
+    -- Select first 3 tuples and fetch a last tuple's position --
+    app:instance001> result, position = read_view1.space.bands:select({}, { limit = 3, fetch_pos = true })
+    ---
+    ...
+
+    app:instance001> result
+    ---
+    - - [1, 'Roxette', 1986]
+      - [2, 'Scorpions', 1965]
+      - [3, 'Ace of Base', 1987]
+    ...
+
+    app:instance001> position
+    ---
+    - kQM
+    ...
+
+    -- Then, you can pass this position as the 'after' parameter --
+    app:instance001> read_view1.space.bands:select({}, { limit = 3, after = position })
+    ---
+    - - [4, 'The Beatles', 1960]
+      - [5, 'Pink Floyd', 1965]
+      - [6, 'The Rolling Stones', 1962]
+    ...
+
+
+
+.. _3-0-iproto_tuples:
+
+IPROTO tuple format
+~~~~~~~~~~~~~~~~~~~
+
+Starting with the 3.0 version, the IPROTO protocol is extended to support for sending names of tuple fields in the :ref:`IPROTO_CALL <box_protocol-call>` and other IPROTO responses.
```

Comment:
"protocol is extended to support for" sounds too complex. Perhaps there is a simpler wording for this.